drivers/net/ksz9477.c

```diff
@@ -211,6 +219,391 @@ static int ksz9477_sgmii_write_indirect(uint32_t address, uint16_t *value,
+
+  ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_DATA(port), reg);
+  if (ret != OK)
+    {
+      goto out;
+    }
+
+  /* Set MMD operation mode */
+
+  if (len <= 1)
+    {
+      /* no post increment */
+
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                                KSZ9477_MMD_OP_MODE_NO_INCREMENT | dev);
+      if (ret != OK)
+        {
+          goto out;
+        }
+    }
+  else
+    {
+      /* post increment on reads and writes */
+
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                                KSZ9477_MMD_OP_MODE_RW_INCREMENT | dev);
+      if (ret != OK)
+        {
+          goto out;
+        }
+    }
+
+  /* Read value */
+
+  while (len-- && ret == OK)
+    {
+      ret = ksz9477_reg_read16(KSZ9477_PHY_MMD_DATA(port), value);
+      value++;
+    }
+
+out:
+
+  return ret;
+}
+
+static int ksz9477_mmd_write_indirect(ksz9477_port_t port, uint8_t dev,
+                                      uint16_t reg, uint16_t *value,
+                                      uint16_t len)
+{
+  int ret;
+
+  /* Set up MMD device address */
+
+  ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                            KSZ9477_MMD_OP_MODE_REGISTER | dev);
+  if (ret != OK)
+    {
+      goto out;
+    }
+
+  /* Select register */
+
+  ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_DATA(port), reg);
+  if (ret != OK)
+    {
+      goto out;
+    }
+
+  /* Set MMD operation mode */
+
+  if (len <= 1)
+    {
+      /* no post increment */
+
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                                KSZ9477_MMD_OP_MODE_NO_INCREMENT | dev);
+      if (ret != OK)
+        {
+          goto out;
+        }
+    }
+  else
+    {
+      /* post increment on reads and writes */
+
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                                KSZ9477_MMD_OP_MODE_RW_INCREMENT | dev);
+      if (ret != OK)
+        {
+          goto out;
+        }
+    }
+
+  /* Write value */
+
+  while (len-- && ret == OK)
+    {
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_DATA(port), *value);
+      value++;
+    }
+
+out:
+
+  return ret;
+}
+
+#if 1
+static int ksz9477_custom_error_fixes(ksz9477_port_t port)
+{
+  int ret = OK;
+  int j;
+  uint16_t regval16;
+  struct errata err1[11];
+  struct errata err2[4];
+  struct errata err3[18];
+  struct errata mod9[12];
+
+  /* First turn off autoneg */
+
+  ret = ksz9477_reg_write16(KSZ9477_PHY_CONTROL(port), 0x2100);
+
+  if (ret != OK)
+    {
+      nerr("PHY Control register write failure, ret %d\n", ret);
+      return ret ? ret : -EINVAL;
+    }
+
+  /* Set Remote Loopback register */
+
+  ret = ksz9477_reg_write16(KSZ9477_PHY_REMOTE_LP(port), 0x00f0);
+  if (ret != OK)
+    {
+      nerr("PHY Remote Loopback register write failure, ret %d\n", ret);
+      return ret ? ret : -EINVAL;
+    }
+
+  /* Needed configurations for MMD Signal Quality */
+
+  err1[0].reg = 0xce;
```

Comment:
These could be directly initialized at the array definition; like "const struct...err1 = { {...}, {...}...};"

Reply:
done